packages/providers/src.ts/base-provider.ts

```diff
@@ -1257,20 +1257,26 @@ export class BaseProvider extends Provider implements EnsProvider {
 
     async getCode(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
         await this.getNetwork();
-        const params = await resolveProperties({
-            address: this._getAddress(addressOrName),
-            blockTag: this._getBlockTag(blockTag)
-        });
-
-        const result = await this.perform("getCode", params);
+        addressOrName = await addressOrName;
+        const { shard, realm, num } = getAccountFromAddress(addressOrName);
+        const shardNum = BigNumber.from(shard).toNumber();
+        const realmNum = BigNumber.from(realm).toNumber();
+        const accountNum = BigNumber.from(num).toNumber();
+        const endpoint = '/api/v1/contracts/' + shardNum + '.' + realmNum + '.' + accountNum;       
         try {
-            return hexlify(result);
+            let { data } = await axios.get(this.mirrorNodeUrl + endpoint);
+            console.log("REST Api call: " + this.mirrorNodeUrl + endpoint);
+            if (data.bytecode != null) {
+                return hexlify(data.bytecode);
+            }
+            return null;
```

Comment:
We need to see what will `getCode` return in Ethereum for addresses that are not contracts (f.e Externally Owned Accounts) and return the same result. F.e if `address` provided is hedera `account` return the same response as if we are querying an address in ethereum that is NOT contract